src/www/ui/page/AdminLicenseCandidate.php

```diff
@@ -262,6 +279,32 @@ private function mergeCandidate($candidate, $suggest, $vars)
     return true;
   }
 
+  protected function doDeleteCandidate($rfPk)
+  {
+    $dbManager = $this->getObject('db.manager');
+    $stmt = __METHOD__.".$rfPk";
+    $dbManager->prepare($stmt, "SELECT DISTINCT(uploadtree_fk) FROM clearing_event "
```

Comment:
it is simpler to read if you use multiline strings and formatting:
```php
   $dbManager->prepare($stmt,
        "SELECT DISTINCT(uploadtree_fk)
           FROM clearing_event 
          WHERE rf_fk=$1
            AND removed=false
            AND uploadtree_fk NOT IN (SELECT DISTINCT(uploadtree_fk)
                                        FROM clearing_event
                                       WHERE rf_fk=$1
                                         AND removed=true)");
```
which can (AFAIK) be simplified to 

```php
   $dbManager->prepare($stmt,
        "SELECT DISTINCT(uploadtree_fk)
           FROM clearing_event 
          WHERE rf_fk=$1
            AND removed=false");
```

Reply:
Candidate licenses not included in any scanner findings. they are used once scanning gets finished.

Reply:
the `reportImport` PR will change this and create candidates as scanner findings.

Reply:
I think we need this then with the report agent